FirebaseCore/Internal/Sources/HeartbeatLogging/RingBuffer.swift

```diff
@@ -38,14 +38,15 @@ struct RingBuffer<Element>: Sequence {
       return nil
     }
 
-    defer {
-      // Increment index, wrapping around to the start if needed.
-      tailIndex += 1
-      tailIndex %= circularQueue.count
-    }
-
     let replaced = circularQueue[tailIndex]
     circularQueue[tailIndex] = element
+
+    // Increment index, wrapping around to the start if needed.
```

Comment:
I can't imagine that `defer` wouldn't be exactly equivalent to this.

Reply:
Yeah, I agree. Unless some weird Swift bug is occurring where some of the `defer` block is executed. Even that seems highly unlikely though

Reply:
Assuming nothing weird is going on with the use of `defer`, this PR is functionally equivalent to the current code. Maybe a little bit easier to read, but that's all.